### ISSUE TITLE
(Question) Installation instructions for beginners

### ISSUE THREAD

**Issue by ByteSharkie** [ikz87/dots-2.0#29]
Good morning dev,

Apologies for opening an issue ticket but I'm kinda lost on how to install it and make it properly work. I'm a new Linux desktop user and I certainly don't know what to do. 

I created a vanilla arch system, installed the yay and run the dependencies list. Then I cloned the rep run the install.sh and run the bspwnrc and autostart but is not working properly. Kitty is not configured, eww is not working properly, the windows animation is not working etc.

Should I install an arch with a KDE plasma or just arch with xorg and bspwm? Should I install more dependencies that I am unaware of? Is there a proper path that should i follow to properly install it?

Any kind of assistance will be much appriciated.

Thank you for your kind efforts.

**Comment by ikz87** [comment]
Hi, thanks for checking out my dots.
Apologies in advance because I do think the installation steps are explained in a way that might confuse less experienced users. Let me tackle every issue you described one by one.

Configurations for kitty should work out of the box, colors and font wise. If what you want is the fetching program and prompt, that's part of my bash configuration which you can find in the personal branch of this repo, [here](https://github.com/ikz87/dots-2.0/blob/personal/.bashrc). 

As for the dependencies, are you sure everything got installed correctly? just in case I'd recommend you run the command again. Also, remember you can use any other package manager of your liking by replacing the `yay` at the start of the command.

You saying that you ran bspwmrc and autostart is a little weird, those files are *not* supposed to be ran manually. When you start you X server (wether it is with `startx` or a [display manager](https://wiki.archlinux.org/title/display_manager)) with a bspwm session (you can refer to my [xinitrc](https://github.com/ikz87/dots-2.0/blob/personal/.xinitrc)), bspwmrc should be automatically run behind the scenes, and then autostart should follow. Making that clear just in case.

Now, there's a little section on the dependency list called "Other", these are packages that can't be found in the AUR, that you will have to install from source (every one of those packages have their own installation process, which I think are easy enough to follow), specifically, what might be causing the animations to not work is not having [FT-Labs-picom](https://github.com/FT-Labs/picom) installed.

Eww not working certainly is the hardest to explain out of everything you said, I'll need more info. I no longer use the `eww-git` package for eww personally, but it shouldn't matter, however, could you show me what's the output of `eww -V` for you? 
Have you tried opening eww windows manually? if so, does it specify any errors?. One thing that could be happening is that the install script *somehow*) didn't work properly, try running eww with the config straight from the repo. Move into the root of the repo on a terminal and run these commands: 
```
killall eww
eww -c ./.config/eww/mybar open bar_left
```
You should see the following happening
![image](https://user-images.githubusercontent.com/98569017/212478113-004e2496-0584-4585-8753-eb46a5c970a7.png)

Please let me know if that worked, and if it didn't, show me the output of whatever command that failed so we can continue to debug further.

As for your last question, excluding all the dependencies explicitly mentioned, only Xorg and BSPWM are needed for the rice. The KDE pacakge mantioned at the end is optional and only helps with media integration with the left bar when using brave.

**Comment by ByteSharkie** [comment]
Hi again, thank you for your swift and detailed response. I have installed a fresh window, will go in details.

Regarding the packages, every dependancy was successful except xwinfo-git
![image](https://user-images.githubusercontent.com/67751081/212489719-6232e19e-4677-41a6-9814-b80bb4123a32.png).

For the xinitrc i copied the template that is provided by default and add the below
`xsetroot -cursor_name left_ptr`
`exec bspwm`

For the .bashrc I didn't proceed with any modifications.

When I `startx` , I'm getting promted to the enviroment, but when i open a terminal (ctrl + alt + t) the window is empty
![image](https://user-images.githubusercontent.com/67751081/212489879-4c92841a-4c48-47e7-94c2-2a0e61f356d1.png)

Therefore, I'm unable to write any command regarding the eww.

As for the [FT-Labs-picom](https://github.com/FT-Labs/picom), I have installed it but it seems to be very very laggy. and the window I open (e.g. file manager) keeps "trempling"
![image](https://user-images.githubusercontent.com/67751081/212490330-43445507-25f5-452a-a779-88c8464142e9.png)

Once again thank you for your time. I honestly believe it's entirely my fault that it is not working due to inexpirience

**Comment by ikz87** [comment]
Seems to me like a problem related to your graphics card, either having to do with Xorg or picom. Try commenting the line that runs picom in `.config/bspwm/autostart`:
![image](https://user-images.githubusercontent.com/98569017/212506774-7291be95-5419-4834-9286-539967c66977.png)
Then start X again, this will make it so you run bspwm without the compositor, that way we can see if it's a compositor setting I'm using that's giving you trouble. 
Also it could be of help if you provided your pc specs, I'm mostly interested in your graphics card, and please, don't think this is your fault, every pc is different and some things will just behave a lot different

**Comment by ikz87** [comment]
Also, a little unrelated but I can see eww itself working fine, it's just some parts of the central bar not working correctly.
This made me realize I forgot a couple dependencies (already fixed them).
Install the `sysstat` package and the CPU usage module should start working.
For the temperature, the `lm_sensors` might be missing (would be weird tho), install it and set it up following this [guide](https://wiki.archlinux.org/title/lm_sensors#Setup), if your pc supports it, the module should start working.
As for the battery level, `acpi` is the command I use to get it, my guess (correct me if I'm wrong) is that you're using a computer that had no battery (like a desktop pc), so it isn't supported.


TL;DR eww is working as intended, the modules are failing because some packages are missing (my fault), run the command below and set up sensors like [this](https://wiki.archlinux.org/title/lm_sensors#Setup)
```
yay -Sy sysstat lm_sensors --needed
```

**Comment by ByteSharkie** [comment]
Hi again, I have disabled the picom and it seems now to be working.

![image](https://user-images.githubusercontent.com/67751081/212529834-e81e1b74-434d-4a1e-b4ba-6bdecaec809e.png)

specs:
CPU: AMD Ryzen 5 3600 6-Core 3600Mhz
RAM: 32GB
GPU: AMD Radeon RX Vega 56

You are correct, I'm in a desktop pc and running it in a vm.

As a side question, are you using any type of configuration for the file manager because on my previous screenshot it seems that is unconfigured.

**Comment by awarewen** [comment]
I also have the same situation about picom. The original picom-animations-git works well on my computer, but FT-Labs picom does not work normally. 

![image](https://user-images.githubusercontent.com/53365857/212536992-621fb08f-0a6b-4584-9d87-3411cbaaa56a.png)

This is what happens when I use peek to record the screen. It seems that there is a layer of fuzzy filter, which makes peek unable to see the desktop clearly:
![image](https://user-images.githubusercontent.com/53365857/212537775-81436f90-5422-4d56-af82-afa237c5d776.png)

![Peek 2023-01-15 19-06](https://user-images.githubusercontent.com/53365857/212537262-837109c9-ffb9-4d0e-af28-da8808e720f1.gif)

![image](https://user-images.githubusercontent.com/53365857/212537186-4a9af70f-4c70-441b-90b8-e64b54768c6c.png)

And this is picom-animations-git:
![Peek 2023-01-15 19-17](https://user-images.githubusercontent.com/53365857/212537576-047d9b20-7d69-45bd-8eb1-91ce80b22efd.gif)

**Comment by awarewen** [comment]
Notebook computer:[ GPD pocket 3](https://gpd.hk/gpdpocket3)
OS: arch linux 
CPU: Intel Core i7-1195G7
 graphics card: Core graphics card

**Comment by ikz87** [comment]
Looking back at my previous picom config, I couldn't figure out why the issues might be happening, so I'm going back to using `picom-animations-git` in the main branch. If any of you find out what setting might be causing this, please let me know as I do think `FT-Labs-Picom` is a better alternative, I just couldn't replicate the problem in my machine

**Comment by ikz87** [comment]
As of now, `picom-animations-git` will again be the picom fork used.
@AstronatShark I recommend you run `yay -Sy picom-animations-git` and update your picom config (you can just pull changes from the repo and run install.sh again if you want). Let me know if this picom works for you

**Comment by ikz87** [comment]
> As a side question, are you using any type of configuration for the file manager because on my previous screenshot it seems that is unconfigured.

Regarding the file manaager, that works with the gtk-theme and icon set.
I'm currently working configuring wpgtk (the program that provides the colorschemes) for it to work well with the gtk themes, might get it done today. 
Will add to the repo when everything is working correctly

**Comment by ikz87** [comment]
Oh, and one more thing. If picom is still laggy for you @AstronatShark it might be due to the fact that you are running linux in a VM. Looking at your specs I can see that you have a really powerful PC, but even then, VM's tend to be no the laggy side when it comes to rendering stuff.
I'd also double check if the graphics card drivers are correctly installed

**Comment by ikz87** [comment]
> As a side question, are you using any type of configuration for the file manager because on my previous screenshot it seems that is unconfigured.

Sorry, this issue thread has become kind of a mess, but this is done now. @AstronatShark I recommend you just pull or clone the repo again, and repeat the reinstall process, should fix a couple of your issues at once. Let me know if it worked

**Comment by ByteSharkie** [comment]
I installed `picom-animations-git` and re applied the `install.sh` but with no improvment. Will try again as you instructed on a fresh vm and will let you know. I thought about the drivers too and the issue didn't resolved. I think that the issue might be with the vm too.

**Comment by ikz87** [comment]
I'll wait for your feedback once you try on a fresh install then, but since I'm 99% positive your problems have to do with the VM or GPU drivers, I'll close the issue afterwards unless there's a hint that something else is causing you trouble.
If it doesn't improve I'd suggest not using picom at all or trying another rice altogether :c

**Comment by awarewen** [comment]
> 再次嗨，感谢您迅速而详细的回复。 我已经安装了一个新窗口，将详细介绍。
> 
> 关于包，除了 xwinfo-git 之外，每个依赖项都是成功的 ![图片](https://user-images.githubusercontent.com/67751081/212489719-6232e19e-4677-41a6-9814-b80bb4123a32.png).
> 
> 对于 xinitrc，我复制了默认提供的模板并添加以下内容 `xsetroot -cursor_name left_ptr` `exec bspwm`
> 
> 对于 .bashrc，我没有进行任何修改。
> 
> 当我 `startx`，我被提示进入环境，但是当我打开一个终端（ctrl + alt + t）时，窗口是空的 ![图片](https://user-images.githubusercontent.com/67751081/212489879-4c92841a-4c48-47e7-94c2-2a0e61f356d1.png)
> 
> 因此，我无法编写任何关于 eww 的命令。
> 
> 至于 [FT-Labs-picom ](https://github.com/FT-Labs/picom) ，我已经安装了它，但它似乎非常滞后。 我打开的窗口（例如文件管理器）一直在“颤抖” ![图片](https://user-images.githubusercontent.com/67751081/212490330-43445507-25f5-452a-a779-88c8464142e9.png)
> 
> Once again thank you for your time. I honestly believe it's entirely my fault that it is not working due to inexpirience

With regard to the performance of FLP(FT-Labs-Picom), you can try to turn off all shadow settings. This is a good choice, because it seems that shadows do not work properly at present. I hope this can help you. After I turn off the shadow setting of FLP, all aspects of a have entered normal operation, but the animation effect of FLP seems to be less than the animation effect of picom-animations.

**Comment by awarewen** [comment]
@ikz87  Hi, bro. Does the shadow setting of fL-Labs-Picom work properly in your environment.

**Comment by ikz87** [comment]
No, i had deactivated them. Were you using FT-Labs-Picom with the old config? when I switched to it I also changed a couple settings. 
See [here](https://github.com/ikz87/dots-2.0/commit/2f16748641d51fb8326d4306f800d1e2fb991716).
Could you please try using that fork again, but with the changed configuration? If there are no issues on your side, I'll bring FLP back for the main branch

**Comment by awarewen** [comment]
> 
FLP worked fine after I turned off the shadow setting, but there seems to be some unknown error output, but it doesn't affect usage for now.
![图片](https://user-images.githubusercontent.com/53365857/212641273-57079e3e-53ec-421a-a11b-0049ab8a5be0.png)

**Comment by awarewen** [comment]
> > 
> 
> FLP worked fine after I turned off the shadow setting, but there seems to be some unknown error output, but it doesn't affect usage for now. ![图片](https://user-images.githubusercontent.com/53365857/212641273-57079e3e-53ec-421a-a11b-0049ab8a5be0.png)
---

---
When the eww - pop music component and closing can print this mistake.But I don't know what this error means.

https://user-images.githubusercontent.com/53365857/212644706-b9ad5998-7d45-486e-a01f-dc7d108bb3b5.mp4

```
[ 01/16/2023 17:15:18.082 win_set_flags ERROR ] Flags set on a destroyed window 0x00a75954 (Eww - music)
[ 01/16/2023 17:15:18.082 win_clear_flags WARN ] Flags cleared on a destroyed window 0x00a75954 (Eww - music)
```
But there is no such error in picom-animations.

https://user-images.githubusercontent.com/53365857/212643199-b8605729-155b-488f-bbe7-364ccd5931b6.mp4

**Comment by ikz87** [comment]
Just to be sure, you're using [this](https://github.com/ikz87/dots-2.0/blob/2f16748641d51fb8326d4306f800d1e2fb991716/.config/picom.conf) config file, right?
If FLP doesn't cause any practical problems for now, I'd appreciate if you could test further an open a new issue if you happen to find some actual problem other than the error messages, because I'm not being able to replicate the bug

**Comment by awarewen** [comment]
> 
Now things have changed a bit, and when I use the configuration you gave me, the aforementioned errors no longer occur.  I think FTP can go back to the main line. He is working very well on my machine at the moment, and we can look for other minor problems and fix them.  :+1: 

![图片](https://user-images.githubusercontent.com/53365857/212647343-6bc31f3a-b43b-4ccb-8477-8718a214b79d.png)


NIIIIIIIICE!!!~~~~~~
```
opacity-rule = [
  "85:class_g = 'firefox' && focused",
  "60:class_g = 'firefox' && !focused",
  "85:class_g = 'kitty' && focused",
  "60:class_g = 'kitty' && !focused",
  "100:fullscreen",
];
```
![图片](https://user-images.githubusercontent.com/53365857/212649391-e534d5ae-c661-46e8-ad8a-283eb438e105.png)

**Comment by ikz87** [comment]
Glad it worked out! :)
Even though it's a bummer I couldn't help you fix your main issue @AstronatShark (picom) due to it being most likely related to external factors to my rice, I wanna thank you for pointing out other very solvable issues with the rice installation process:
- Missing dependencies (sysstat and lm_sensors, needed for some bar modules to work) *FIXED* 
- GTK themes not working with the colorscheme (what you said about the file manager not being properly configured) *FIXED*

Also with the help of @awarewen I think we can confirm FLP is a viable option (and dare I say better than picom-animations-git) so I'm reverting this [commit](https://github.com/ikz87/dots-2.0/commit/b5fc495debe3e7e9af91518e6118acd68b7f2a74)

I suggest both of you run `install.sh` on the latest version of the repo to get the latest changes.
Will close the issue now, and again, thank you :)
